core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -134,11 +134,12 @@ public void createTable(
     ) {
         CharSequence lockedReason = lock(securityContext, struct.getTableName(), "createTable");
         if (null == lockedReason) {
-            if (writerPool.exists(struct.getTableName())) {
-                throw EntryUnavailableException.instance("table exists");
-            }
             boolean newTable = false;
             try {
+                if (getStatus(securityContext, path, struct.getTableName()) != TableUtils.TABLE_DOES_NOT_EXIST) {
```

Comment:
this check was originally outside of the try/finally block, but this was leaking locks. 
this was hidden visible, because the old impl never throw the exception anyway.